src/pages/patientView/structuralVariant/column/AnnotationColumnFormatter.tsx

```diff
@@ -0,0 +1,177 @@
+
+export default class AnnotationColumnFormatter {
+    public static getData(
+        structuralVariantData: StructuralVariant[] | undefined,
+        oncoKbCancerGenes?: RemoteData<CancerGene[] | Error | undefined>,
+        oncoKbData?: RemoteData<IOncoKbData | Error | undefined>,
+        usingPublicOncoKbInstance?: boolean,
+        uniqueSampleKeyToTumorType?: { [sampleId: string]: string },
+        studyIdToStudy?: { [studyId: string]: CancerStudy }
+    ) {
+        let value: IAnnotation;
+
+        if (structuralVariantData) {
+            let oncoKbIndicator: IndicatorQueryResp | undefined = undefined;
+            let oncoKbStatus: IAnnotation['oncoKbStatus'] = 'complete';
+
+            let oncoKbGeneExist = false;
+            let isOncoKbCancerGene = false;
+            if (oncoKbCancerGenes && !(oncoKbCancerGenes instanceof Error)) {
+                oncoKbGeneExist = _.some(
+                    oncoKbCancerGenes.result,
+                    (gene: CancerGene) =>
+                        gene.oncokbAnnotated &&
+                        (gene.entrezGeneId ===
+                            structuralVariantData[0].site1EntrezGeneId ||
+                            gene.entrezGeneId ===
+                                structuralVariantData[0].site2EntrezGeneId)
+                );
+                isOncoKbCancerGene = _.some(
+                    oncoKbCancerGenes.result,
+                    (gene: CancerGene) =>
+                        gene.entrezGeneId ===
+                            structuralVariantData[0].site1EntrezGeneId ||
+                        gene.entrezGeneId ===
+                            structuralVariantData[0].site2EntrezGeneId
+                );
+            }
+
+            // oncoKbData may exist but it might be an instance of Error, in that case we flag the status as error
+            if (oncoKbData && oncoKbData.result instanceof Error) {
+                oncoKbStatus = 'error';
+            } else if (oncoKbGeneExist) {
+                // actually, oncoKbData.result shouldn't be an instance of Error in this case (we already check it above),
+                // but we need to check it again in order to avoid TS errors/warnings
+                if (
+                    oncoKbData &&
+                    oncoKbData.result &&
+                    !(oncoKbData.result instanceof Error) &&
+                    oncoKbData.isComplete
+                ) {
+                    oncoKbIndicator = this.getIndicatorData(
+                        structuralVariantData,
+                        oncoKbData.result,
+                        uniqueSampleKeyToTumorType,
+                        studyIdToStudy
+                    );
+                }
+                oncoKbStatus = oncoKbData ? oncoKbData.status : 'pending';
+            }
+
+            value = {
+                hugoGeneSymbol: structuralVariantData[0].site1HugoSymbol, //TODO: which site hugo gene symbol should be used?
```

Comment:
We probably need to redesign the interface for annotation, the oncokb card content needs to updated to since we need to validate both genes are not in oncokb.

Reply:
But for now, we should check both sites and pick the one with higher order. geneIsAnnotated>geneIsCancerGene>geneNotCancerGene